lib/rules/no-extra-parens.js

```diff
@@ -169,6 +169,26 @@ module.exports = {
             return ruleApplies(node) && isParenthesisedTwice(node);
         }
 
+        /**
+         * Determines if a node that is expected to be parenthesised is surrounded by
```

Comment:
Do you mind expanding on what `considering precedence` means here? Alternatively, a comment below (with code examples) would also be helpful. Specifically, explaining why `isParenthesisedTwice(node)` is being checked here would help clarify the behavior for the next person!

Reply:
@kaicataldo 
yes! it makes sense. thanks! I changed it.